src/main/java/org/georchestra/gateway/filter/headers/HeaderProvider.java

```diff
@@ -18,17 +18,65 @@
  */
 package org.georchestra.gateway.filter.headers;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.function.Consumer;
+import java.util.stream.Collectors;
 
+import org.georchestra.gateway.filter.headers.providers.UserSecurityHeadersProvider;
+import org.springframework.core.Ordered;
 import org.springframework.http.HttpHeaders;
+import org.springframework.security.config.web.server.ServerHttpSecurity;
 import org.springframework.web.server.ServerWebExchange;
 
+import lombok.extern.slf4j.Slf4j;
+
 /**
- * Extension point for contributors to HTTP request headers sent to back-end
- * services.
+ * Extension point to aid {@link AddSecHeadersGatewayFilterFactory} in appending
+ * the required HTTP request headers to proxied requests.
+ * <p>
+ * Beans of this type are strategy objects that contribute zero or more HTTP
+ * request headers to be appended to proxied requests to back-end services.
+ * 
+ * @see UserSecurityHeadersProvider
  */
-public interface HeaderProvider {
+@Slf4j
+public abstract class HeaderProvider implements Ordered {
```

Comment:
is it necessary to have the HeaderProvider being ordered ? Once the request to the proxified webapps will be performed, I cannot see where the order will have an importance ?

Reply:
I guess I got it actually: we probably want to have the provider which sanitizes the request / removes the sec-*, before the one which adds them

Reply:
doesn't hurt, just in case we need to override/sanitize